tests/wasmfs/wasmfs_stat.c

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright 2021 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and the
+ * University of Illinois/NCSA Open Source License.  Both these licenses can be
+ * found in the LICENSE file.
+ */
+
+#include <assert.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+// FIXME: Individual test to verify fstat in isolation. May get merged with
+// others later.
+
+int main() {
+  // Attempt to call fstat on an invalid fd.
+  errno = 0;
+  struct stat invalid;
+  int result = fstat(-1, &invalid);
+  assert(result == -1);
+  assert(errno == EBADF);
+
+  // Test opening a file and calling fstat.
+  struct stat file;
+  int fd = open("/dev/stdout/", O_RDONLY);
+  fstat(fd, &file);
+  assert(fd >= 0);
+
+  off_t fileInode = file.st_ino;
+
+  assert(file.st_size == 0);
+#ifdef WASMFS
+  assert((file.st_mode & S_IFMT) == S_IFREG);
+#else
+  assert((file.st_mode & S_IFMT) == S_IFCHR);
+#endif
+  assert(file.st_dev);
+  assert(file.st_nlink);
+  assert(file.st_uid == 0);
+  assert(file.st_gid == 0);
+  assert(file.st_rdev);
+  assert(file.st_blocks == 0);
+  assert(file.st_blksize == 4096);
+#ifdef WASMFS
+  assert(file.st_atim.tv_sec == 0);
+  assert(file.st_mtim.tv_sec == 0);
+  assert(file.st_ctim.tv_sec == 0);
+#endif
```

Comment:
It's worth at least a TODO comment here about unifying these expectations. I'm not sure it's worth testing the times if we consider those an incomplete feature.

Reply:
Removed for now